backend/tracim_backend/lib/core/content.py

```diff
@@ -1070,10 +1071,24 @@ def _get_all_query(
         :return: Query object
         """
         order_by_properties = order_by_properties or []  # FDV
-        assert parent_id is None or isinstance(parent_id, int)
+        assert not parent_ids or isinstance(parent_ids, list)
         assert content_type_slug is not None
+        assert not complete_path_to_id or isinstance(complete_path_to_id, int)
         resultset = self._base_query(workspace)
 
+        # INFO - G.M - 2018-11-12 - Get list of all ancestror
+        #  of content, workspace root included
+        if complete_path_to_id:
+            content = self.get_one(complete_path_to_id, content_type_list.Any_SLUG)
+            if content.parent_id:
+                content = content.parent
+                while content.parent_id:
+                    parent_ids.append(content.content_id)
+                    content = content.parent
+                parent_ids.append(content.content_id)
+                # TODO - G.M - 2018-11-12 - add workspace root to parent_id list
+                parent_ids.append(0)
```

Comment:
Why add `0` to parent_ids ? My interpretation is "By default, add contents who are not in folder". It is for retro-compatibility ? Why do this only if `if complete_path_to_id` and `if content.parent_id:` ?

Reply:
It's about complete_path_to_id feature, this feature add an easy way to say "i want to filter by all parent of this content_id". root is the parent of a content which as no more parent, we should add it to have complete content tree from content to root.

Reply:
> It's about complete_path_to_id feature, this feature add an easy way to say "i want to filter by all parent of this content_id".

We do not want to "filter by all parent of the content id" but we want to include all contents which have a parent being in the list of parent_ids.

> root is the parent of a content which as no more parent, we should add it to have complete content tree from content to root.

Contents at root level do not have parents. If we want to get them, the natural way to do this would be "parent id is in parent_ids or is None". adding "0" seems to be natural.

By the way, I believe parent_ids.append(0) should always be added if `complete_path_to_id` is set

Reply:
> We do not want to "filter by all parent of the content id" but we want to include all contents which have a parent being in the list of parent_ids.

Yes, in fact when is speak about filtering by parent, i speak about including those content.

> By the way, I believe parent_ids.append(0) should always be added if complete_path_to_id is set

you're right.